gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -638,242 +662,292 @@ _gst_tensor_filter_transform_validate (GstBaseTransform * trans,
+      trans_data->outbuf);
   if (retval != GST_FLOW_OK)
     return retval;
 
-  allocate_in_invoke = gst_tensor_filter_allocate_in_invoke (priv);
+  trans_data->allocate_in_invoke = gst_tensor_filter_allocate_in_invoke (priv);
 
-  in_flexible =
+  trans_data->in_flexible =
       gst_tensor_pad_caps_is_flexible (GST_BASE_TRANSFORM_SINK_PAD (trans));
-  out_flexible =
+  trans_data->out_flexible =
       gst_tensor_pad_caps_is_flexible (GST_BASE_TRANSFORM_SRC_PAD (trans));
 
-  if (priv->prop.invoke_dynamic && !out_flexible) {
+  if (priv->prop.invoke_dynamic && !trans_data->out_flexible) {
     ml_loge
         ("Dynamic Invoke of tensor filter is activated but the output of tensor filter is static tensors. Currently, only flexible tensors is supported as output of dynamic invoke. If you don't want to dynamic invoke, remove the invoke-dynamic option of tensor filter.");
     return GST_FLOW_ERROR;
   }
 
-  /* 1. Get all input tensors from inbuf. */
-  /* Internal Logic Error or GST Bug (sinkcap changed!) */
-  num_tensors = gst_tensor_buffer_get_count (inbuf);
+  return retval;
+}
+
+/**
+ * @brief Internal function to get input tensors.
+ */
+static GstFlowReturn
+gst_tensor_filter_transform_get_input_tensors (GstBaseTransform * trans,
+    FilterTransformData * trans_data)
+{
+  GstTensorFilter *self = GST_TENSOR_FILTER_CAST (trans);
+  GstTensorFilterPrivate *priv = &self->priv;
+  GstTensorFilterProperties *prop = &priv->prop;
+  guint i;
+  gsize hsize;
+
+  trans_data->in_num_tensors = gst_tensor_buffer_get_count (trans_data->inbuf);
 
-  for (i = 0; i < num_tensors; i++) {
-    in_mem[i] = gst_tensor_buffer_get_nth_memory (inbuf, i);
-    if (!gst_memory_map (in_mem[i], &in_info[i], GST_MAP_READ)) {
+  for (i = 0; i < trans_data->in_num_tensors; i++) {
+    trans_data->in_mem[i] =
+        gst_tensor_buffer_get_nth_memory (trans_data->inbuf, i);
+    if (!gst_memory_map (trans_data->in_mem[i], &trans_data->in_info[i],
+            GST_MAP_READ)) {
       ml_logf_stacktrace
           ("gst_tensor_filter_transform: For the given input buffer, tensor-filter (%s : %s) cannot map input memory from the buffer for reading. The %u-th memory chunk (%u-th tensor) has failed for memory map.\n",
           prop->fwname, TF_MODELNAME (prop), i, i);
-      goto mem_map_error;
+      return GST_FLOW_ERROR;
     }
 
     hsize = 0;
-    if (in_flexible) {
-      gst_tensor_meta_info_parse_header (&in_meta[i], in_info[i].data);
-      hsize = gst_tensor_meta_info_get_header_size (&in_meta[i]);
-      gst_tensor_meta_info_convert (&in_meta[i], &prop->input_meta.info[i]);
+    if (trans_data->in_flexible) {
+      gst_tensor_meta_info_parse_header (&trans_data->in_meta[i],
+          trans_data->in_info[i].data);
+      hsize = gst_tensor_meta_info_get_header_size (&trans_data->in_meta[i]);
+      gst_tensor_meta_info_convert (&trans_data->in_meta[i],
+          &prop->input_meta.info[i]);
```

Comment:
The current code is fine, but what if we make this part into a function like below?
```
static gsize
gst_tensor_filter_get_header_size (GstTensorFilterPrivate * priv, FilterTransformData * trans_data)
{
  gsize header_size = 0;
  GstTensorFilterProperties *prop = &priv->prop;
  if (trans_data->in_flexible) {
      gst_tensor_meta_info_parse_header (&trans_data->in_meta[I], trans_data->in_info[i].data);
      header_size = gst_tensor_meta_info_get_header_size (&trans_data->in_meta[i]);
      gst_tensor_meta_info_convert (&trans_data->in_meta[I], &prop->input_meta.info[i]);
  }
  return header_size;
}  
```